Solutions/Corvus.Monitoring.AspnetCore.Mvc.Specs/Fakes/TestOperationInstance.cs

```diff
@@ -0,0 +1,82 @@
+﻿// <copyright file="TestOperationInstance.cs" company="Endjin Limited">
+// Copyright (c) Endjin Limited. All rights reserved.
+// </copyright>
+
+namespace Corvus.Monitoring.AspnetCore.Mvc.Specs.Fakes
+{
+    using System.Collections.ObjectModel;
+    using Corvus.Monitoring.Instrumentation;
+
+    public class TestOperationInstance : IOperationInstance
+    {
+        public TestOperationInstance(string name, AdditionalInstrumentationDetail? additionalDetail = null)
+        {
+            this.Name = name;
+
+            if (additionalDetail?.MetricsIfPresent != null)
+            {
+                foreach (KeyValuePair<string, double> metric in additionalDetail.Metrics)
+                {
+                    this.Metrics.Add(metric);
+                }
+            }
+
+            if (additionalDetail?.PropertiesIfPresent != null)
+            {
+                foreach (KeyValuePair<string, string> property in additionalDetail.Properties)
+                {
+                    this.Properties.Add(property);
+                }
+            }
+        }
+
+        public IDictionary<string, double> Metrics { get; private set; } = new Dictionary<string, double>();
+
+        public IDictionary<string, string> Properties { get; private set; } = new Dictionary<string, string>();
+
+        public string Name { get; }
+
+        public bool IsDisposed { get; private set; } = false;
+
+        public DateTimeOffset StartDateTimeUtc { get; } = DateTimeOffset.UtcNow;
+
+        public DateTimeOffset? EndDateTimeUtc { get; private set; }
+
+        public TimeSpan Duration
+        {
+            get
+            {
+                if (!this.IsDisposed)
+                {
+                    throw new InvalidOperationException("Cannot get the duration of a TestOperationInstance until it has been disposed");
```

Comment:
That's unusual! Normally we throw if you try to use something when it has been disposed!

I guess this is only for internal use by the test, so it's not a big deal. It did cause me some head-scratching, but if it's a PITA to do it any other way, then fair enough.

Reply:
Yes, it's because I know that these things should only ever be disposed once so I want things to break under test if it's not working in that way.

That said, I ended up not using this duration property, so I'm just going to delete it.